From d4b988a424cd3ff664795e3fe95a427c8cceb222 Mon Sep 17 00:00:00 2001
From: Colin Guthrie <colin@mageia.org>
Date: Sat, 24 Mar 2012 23:13:25 +0000
Subject: [PATCH 504/509] mageia(not-upstream): Add mount+automount units for
 /proc/bus/usb

This is just a temporary fix until draknetcenter (or lsdetect)
can be refactored to not use /proc/bus/usb

https://bugs.mageia.org/show_bug.cgi?id=4500
---
 Makefile.am                             | 15 +++++++++++++++
 src/core/mount-setup.c                  |  2 --
 units/mageia/proc-bus-usb-setup.service | 19 +++++++++++++++++++
 units/mageia/proc-bus-usb.automount     | 15 +++++++++++++++
 units/mageia/proc-bus-usb.mount         | 16 ++++++++++++++++
 5 files changed, 65 insertions(+), 2 deletions(-)
 create mode 100644 units/mageia/proc-bus-usb-setup.service
 create mode 100644 units/mageia/proc-bus-usb.automount
 create mode 100644 units/mageia/proc-bus-usb.mount

diff --git a/Makefile.am b/Makefile.am
index 693be43..11b054c 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -418,10 +418,25 @@ if TARGET_MAGEIA
 dist_systemunit_DATA += \
 	units/mageia/prefdm.service \
 	units/mageia/display-manager-failure.service \
+	units/mageia/proc-bus-usb-setup.service \
+	units/mageia/proc-bus-usb.mount \
+	units/mageia/proc-bus-usb.automount \
 	units/fedora/rc-local.service \
 	units/fedora/halt-local.service
 systemgenerator_PROGRAMS += \
 	systemd-rc-local-generator
+
+mageia-units-install-data-hook:
+	$(MKDIR_P) -m 0755 \
+		$(DESTDIR)$(systemunitdir)/sysinit.target.wants
+	( cd $(DESTDIR)$(systemunitdir)/sysinit.target.wants && \
+		rm -f proc-bus-usb-setup.service \
+			proc-bus-usb.automount && \
+		$(LN_S) ../proc-bus-usb-setup.service proc-bus-usb-setup.service && \
+		$(LN_S) ../proc-bus-usb.automount proc-bus-usb.automount )
+
+INSTALL_DATA_HOOKS += \
+	mageia-units-install-data-hook
 endif
 
 dist_doc_DATA = \
diff --git a/src/core/mount-setup.c b/src/core/mount-setup.c
index e86a893..f8a7954 100644
--- a/src/core/mount-setup.c
+++ b/src/core/mount-setup.c
@@ -81,8 +81,6 @@ static const char ignore_paths[] =
         /* Legacy cgroup mount points */
         "/dev/cgroup\0"
         "/cgroup\0"
-        /* Legacy kernel file system */
-        "/proc/bus/usb\0"
         /* Container bind mounts */
         "/proc/sys\0"
         "/dev/console\0"
diff --git a/units/mageia/proc-bus-usb-setup.service b/units/mageia/proc-bus-usb-setup.service
new file mode 100644
index 0000000..edaeb20
--- /dev/null
+++ b/units/mageia/proc-bus-usb-setup.service
@@ -0,0 +1,19 @@
+#  This file is part of systemd.
+#
+#  systemd is free software; you can redistribute it and/or modify it
+#  under the terms of the GNU General Public License as published by
+#  the Free Software Foundation; either version 2 of the License, or
+#  (at your option) any later version.
+
+[Unit]
+Description=Legacy USB proc File System Setup
+DefaultDependencies=no
+Conflicts=shutdown.target
+After=systemd-readahead-collect.service systemd-readahead-replay.service
+Before=sysinit.target shutdown.target proc-bus-usb.automount
+ConditionPathExists=!/proc/bus/usb
+
+[Service]
+Type=oneshot
+RemainAfterExit=yes
+ExecStart=-/sbin/modprobe usbcore
diff --git a/units/mageia/proc-bus-usb.automount b/units/mageia/proc-bus-usb.automount
new file mode 100644
index 0000000..72c9111
--- /dev/null
+++ b/units/mageia/proc-bus-usb.automount
@@ -0,0 +1,15 @@
+#  This file is part of systemd.
+#
+#  systemd is free software; you can redistribute it and/or modify it
+#  under the terms of the GNU General Public License as published by
+#  the Free Software Foundation; either version 2 of the License, or
+#  (at your option) any later version.
+
+[Unit]
+Description=Legacy USB proc File System Automount Point
+DefaultDependencies=no
+Before=sysinit.target
+ConditionPathExists=/proc/bus/usb
+
+[Automount]
+Where=/proc/bus/usb
diff --git a/units/mageia/proc-bus-usb.mount b/units/mageia/proc-bus-usb.mount
new file mode 100644
index 0000000..084b306
--- /dev/null
+++ b/units/mageia/proc-bus-usb.mount
@@ -0,0 +1,16 @@
+#  This file is part of systemd.
+#
+#  systemd is free software; you can redistribute it and/or modify it
+#  under the terms of the GNU General Public License as published by
+#  the Free Software Foundation; either version 2 of the License, or
+#  (at your option) any later version.
+
+[Unit]
+Description=Legacy USB proc File System
+DefaultDependencies=no
+
+[Mount]
+What=usbfs
+Where=/proc/bus/usb
+Type=usbfs
+Options=devmode=0664,devgid=43
-- 
1.7.11.4

